### DISCUSSION TITLE
How to test

### DISCUSSION THREAD

**Discussion by tomerslice** [dolanmiu/docx#2169] [Q&A] [answered]
Hello,

I want to add a test for my function and check we are giving all the needed parameters to a template.
How can I read the generated docx to check it, or any other way?
I think that comparing buffer is not working because of timestamps in it.

Thanks.

**Comment by dolanmiu** [comment, marked as answer]
What do you mean by test? Test if the word document is valid? If so, it is possible to test it against the OOXML spec.

It is what `docx` does. Some form of E2E testing I guess:

https://github.com/dolanmiu/docx/blob/master/.github/workflows/demos.yml

**Reply by tomerslice** [reply]
I will try to do that
